chore: see description (SCRUM-65)

DevEquinox opened the pull request:
I'm afraid of breaking the merge queue with a long title.

Implement mock API, and migrate the API into App Container useContext in pages to not recreate an ApiWrapper
Add extension of typescript's Window to allow storing an important variable Remove incorrect page test (anything UI related is through playwright. Speaking of, see future comment in related issue). Configure test projects so we can now run a mock version and a live (for integration tests) version of our project Added separate folders for mock and integration tests Change npm test and the husky test to only run the mock version (since backend is unavailable) Add app.d.ts to tsconfig so the contents (like interface) are recognized

DevEquinox committed:
chore: add mock API, add playwright projects, separate test folders, remove incorrect test, update husky and npm (which affects Github workflow) so they only run the mock version, and add missing global file (SCRUM-65)

Implement mock API, and migrate the API into App Container
useContext in pages to not recreate an ApiWrapper
Add extension of typescript's Window to allow storing an important variable
Remove incorrect page test (anything UI related is through playwright. Speaking of, see future comment in related issue).
Configure test projects so we can now run a mock version and a live (for integration tests) version of our project
Added separate folders for mock and integration tests
Change npm test and the husky test to only run the mock version (since backend is unavailable)
Add app.d.ts to tsconfig so the contents (like interface) are recognized

DevEquinox commented:
Nota: lo del ENV me lo dió chatgpt después de unas cuantas horas jajajaj
Esa fue la única parte que no fue mia (y fue la más tardada)

DanaGtzDev reviewed (dismissed):
LGTM

DevEquinox commented:
Pull request is missing the 5 reviewer requirement. Closing and reopening.

edgarolg reviewed (dismissed):
All good!

LouArcTec reviewed (dismissed):
Love the solution; need to keep this in mind for future integration tests

TbzFlrkd reviewed (dismissed):
Thank you for your hard work!

leyva-bq reviewed (dismissed):
lgtm 👍 love the implemented tests, will keep in mind for future development

DevEquinox commented:
wait
I think there's a little change that's helping a LOT with test loading time (it's gotten a bit unbearable lately).

DevEquinox committed:
refactor: murder the Icon component (SCRUM-65)

The Icon component was importing all lucide icons with import * from, which had an impact on the loading and testing time. On top of that, manual importing resulting in a bit of a redundant check.
Also add an additional config file to tsconfig.json
